theano/gof/opt.py

```diff
@@ -1542,9 +1542,11 @@ def tracks(self):
         return [self.op]
 
     def transform(self, node):
-        if node.op != self.op:
-            return False
-        return node.inputs
+        if inspect.isclass(self.op):
+            if isinstance(node.op, self.op):
+                return node.inputs
+        elif node.op == self.op:
+            return node.inputs
```

Comment:
Maybe should this method return `False` by default (if these both checkings failed) ?

Reply:
By default, all python method return "None" if there is no return statement. Theano will confider this equivalent to False. So it is not requested to return explicitly False. This is done in many optimization already, so is a well tested behavior.

Reply:
Ok !

Reply:
Could this affect things other than ViewOp?  I'm not comfortable making this change close to the release.  However it can go in after.